shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -166,6 +175,28 @@ private void popSystemNavigator() {
         mActivity.finish();
     }
 
+    private JSONObject getClipboardData(String format) throws JSONException {
+        ClipboardManager clipboard = (ClipboardManager) mActivity.getSystemService(Context.CLIPBOARD_SERVICE);
+        ClipData clip = clipboard.getPrimaryClip();
+        if (clip == null)
+            return null;
+
+        if ((format == null || format.equals(kTextPlainFormat)) &&
+            clip.getDescription().hasMimeType(ClipDescription.MIMETYPE_TEXT_PLAIN)) {
+              JSONObject result = new JSONObject();
+              result.put("text", clip.getItemAt(0).getText().toString());
+              return result;
+        }
+
+        return null;
+    }
+
+    private void setClipboardData(JSONObject data) throws JSONException {
+        ClipboardManager clipboard = (ClipboardManager) mActivity.getSystemService(Context.CLIPBOARD_SERVICE);
+        ClipData clip = ClipData.newPlainText("text label?", data.getString("text"));
```

Comment:
Maybe just have `data.GetString("text")` as the user label as well. `"text label?"` does not seem very descriptive.